content/content.go

```diff
@@ -76,6 +89,17 @@ func (s *RulesWithContentStorage) GetRuleContent(ruleID types.RuleID) (*types.Ru
 	return res, found
 }
 
+// GetContentForRecommendation returns content for rule with error key
+func (s *RulesWithContentStorage) GetContentForRecommendation(
+	ruleID types.RuleID,
+) (*local_types.RuleWithContent, bool) {
+	s.RLock()
+	defer s.RUnlock()
```

Comment:
I'm not 100%, but won't it be better to use `sync.Map`? https://golangforall.com/en/post/sync-map.html